cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/MeasurementEndpointInput.java

```diff
@@ -0,0 +1,268 @@
+            this.maxNum = maxNum;
+            return this;
+        }
+
+        public GetAll withBegin(Instant minDate) {
+            this.begin = minDate;
+            return this;
+        }
+
+        public GetAll withEnd(Instant maxDate) {
+            this.end = maxDate;
+            return this;
+        }
+
+        public GetAll withTimezone(String timezone) {
+            this.timezone = timezone;
+            return this;
+        }
+
+        public GetAll withMinHeight(Number minHeight) {
+            this.minHeight = minHeight;
+            return this;
+        }
+
+        public GetAll withMaxHeight(Number maxHeight) {
+            this.maxHeight = maxHeight;
+            return this;
+        }
+
+        public GetAll withMinFlow(Number minFlow) {
+            this.minFlow = minFlow;
+            return this;
+        }
+
+        public GetAll withMaxFlow(Number maxFlow) {
+            this.maxFlow = maxFlow;
+            return this;
+        }
+
+        public GetAll withAgency(String agency) {
+            this.agency = agency;
+            return this;
+        }
+
+        public GetAll withQuality(String quality) {
+            this.quality = quality;
+            return this;
+        }
+
+        @Override
+        protected HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryParameter(LOCATION_QUERY_PARAMETER, locationId)
+                    .addQueryParameter(UNIT_SYSTEM_QUERY_PARAMETER, unitSystem)
+                    .addQueryParameter(MIN_NUMBER_QUERY_PARAMETER, getNullableFieldString(minNum))
+                    .addQueryParameter(MAX_NUMBER_QUERY_PARAMETER, getNullableFieldString(maxNum))
+                    .addQueryParameter(BEGIN_QUERY_PARAMETER, getNullableFieldString(minHeight))
+                    .addQueryParameter(END_QUERY_PARAMETER, getNullableFieldString(maxHeight))
+                    .addQueryParameter(MIN_HEIGHT_QUERY_PARAMETER, getNullableFieldString(minHeight))
+                    .addQueryParameter(MAX_HEIGHT_QUERY_PARAMETER, getNullableFieldString(maxHeight))
+                    .addQueryParameter(MIN_FLOW_QUERY_PARAMETER, getNullableFieldString(minFlow))
+                    .addQueryParameter(MAX_FLOW_QUERY_PARAMETER, getNullableFieldString(maxFlow))
+                    .addQueryParameter(BEGIN_QUERY_PARAMETER, getNullableFieldString(begin))
+                    .addQueryParameter(END_QUERY_PARAMETER, getNullableFieldString(end))
+                    .addQueryParameter(TIMEZONE_QUERY_PARAMETER, timezone)
+                    .addQueryParameter(AGENCY_QUERY_PARAMETER, agency)
+                    .addQueryParameter(QUALITY_QUERY_PARAMETER, quality)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    private static <T> String getNullableFieldString(T field)
```

Comment:
seems useful to put in a base class for broad usage

Reply:
moved to base class